src/lib/libmapfs.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @license
+ * Copyright 2025 The Emscripten Authors
+ * SPDX-License-Identifier: MIT
+ */
+
+addToLibrary({
+  $MAPFS__deps: ['$stringToUTF8OnStack', 'wasmfs_create_map_backend', 'wasmfs_map_create_manifest', 'wasmfs_map_add_to_manifest'],
+  $MAPFS: {
+    createBackend(opts) {
+      return withStackSave(() => {
```

Comment:
The `withStackSave` can be on the `_wasmfs_map_add_to_manifest` call. That will reuse the stack each time, allowing a lower stack size to work in more cases.